rohmu/object_storage/config.py

```diff
@@ -83,10 +89,27 @@ class AzureObjectStorageConfig(StorageModel):
     account_key: Optional[str] = Field(None, repr=False)
     sas_token: Optional[str] = Field(None, repr=False)
     prefix: Optional[str] = None
+    is_secure: bool = True
+    host: Optional[str] = None
+    port: Optional[int] = None
     azure_cloud: Optional[str] = None
     proxy_info: Optional[ProxyInfo] = None
     storage_type: Literal[StorageDriver.azure] = StorageDriver.azure
 
+    @root_validator
+    @classmethod
+    def host_and_port_must_be_set_together(cls, values: Dict[str, Any]) -> Dict[str, Any]:
+        if not values["host"] or not values["port"]:
```

Comment:
This rejects (existing) configs where host and port are both missing, which is not correct. I think you want
```
(values["host"] is None) != (values["port"] is None)
```

Reply:
Yeah, `test_azure_config_host_port_set_together` or a similar test should also verify that the valid cases do not raise an error.

Reply:
Oh, good catch! I added tests to validate the minimal case and fix this.